### PR TITLE
NIAD-574: Mock LDAP response

### PR DESCRIPTION
**Pull request by bartek-sarul**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

**Review by michael-kainos — CHANGES_REQUESTED** [review]
Can we add a short readme of how to use this with docker locally.

File: docker/sds-api/spine-directory-service/sds/sds-mock-ldap.md [hunk 1]
```diff
@@ -1,52 +1,17 @@
 ## SDS - Spine Route Lookup component - Mock LDAP
 
-Spine Route Lookup component has a built in LDAP mocking capability
-To be able to use it, three json configuration files has to be present:
+SDS component has a built in LDAP mocking capability configurable via environment variables:
 
-- server_info.json
-- server_schema.json
-- server_entries.json
+- `SDS_MOCK_LDAP_RESPONSE` - turns the behaviour on and off (`"True"` / `"False"`)
```
Comment (michael-kainos) [inline docker/sds-api/spine-directory-service/sds/sds-mock-ldap.md:5]: can we add these mock specific variables to the docker-compose.yml with defaults or in a override yaml file?

**Review by michael-kainos — CHANGES_REQUESTED** [review]

File: docker/sds-api/spine-directory-service/sds/lookup/mock_data/sds_response.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  "nhsMhsFQDN": "192.168.128.11",
+  "nhsMHsIN": "REPC_IN150016UK05",
+  "nhsMHSPartyKey": "YES-0000806",
+  "nhsMHSActor": [
+    "urn:oasis:names:tc:ebxml-msg:actor:toPartyMSH"
+  ],
+  "nhsMHSPersistDuration": "PT5M",
+  "nhsProductKey": "999",
+  "nhsMHSEndPoint": [
+    "https://192.168.128.11/reliablemessaging/reliablerequest"
```
Comment (michael-kainos) [inline docker/sds-api/spine-directory-service/sds/lookup/mock_data/sds_response.json:25]: should we have ip address here?

File: docker/sds-api/spine-directory-service/sds/lookup/tests/test_routing_reliability.py [hunk 3]
```diff
@@ -1,86 +1,53 @@
...
 PARTY_KEY = " AP4RTY-K33Y"
 INTERACTION_ID = "urn:nhs:names:services:psis:MCCI_IN010000UK13"
 ODS_CODE = "ODSCODE1"
 
-EXPECTED_ROUTING = {
-    'nhsMhsFQDN': 'vpn-client-1411.opentest.hscic.gov.uk',
+EXPECTED_ROUTING_AND_RELIABILITY = {
+    'nhsMHSAckRequested': 'always',
+    'nhsMHSDuplicateElimination': 'always',
     'nhsMHSEndPoint': ['https://vpn-client-1411.opentest.hscic.gov.uk/'],
```
Comment (michael-kainos) [inline docker/sds-api/spine-directory-service/sds/lookup/tests/test_routing_reliability.py:16]: is this safe here?

**Review by Broom94 — APPROVED** [review]

**Comment by Sourceless** [comment]
Force merging due to build not reporting done to github

**Pull request merged**